src/core-common/src/main/java/org/apache/kylin/common/PropertiesDelegate.java

```diff
@@ -39,34 +51,84 @@ public class PropertiesDelegate extends Properties {
     @EqualsAndHashCode.Include
     private final transient IExternalConfigLoader configLoader;
 
+    private final Map delegation;
+
     public PropertiesDelegate(Properties properties, IExternalConfigLoader configLoader) {
-        if(configLoader != null) this.properties.putAll(configLoader.getProperties());
         this.properties.putAll(properties);
         this.configLoader = configLoader;
+        if (configLoader == null) {
+            this.delegation = this.properties;
+        } else if (configLoader instanceof KylinExternalConfigLoader) {
+            this.delegation = new CompositeMapView(((ICachedExternalConfigLoader)this.configLoader).getPropertyEntries(), this.properties);
+        } else if (configLoader instanceof NacosExternalConfigLoader) {
+            this.delegation = new CompositeMapView(((ICachedExternalConfigLoader)this.configLoader).getPropertyEntries(), this.properties);
```

Comment:
Can this configloader cast into ICachedExternalConfigLoader if it is instanceof NacosExternalConfigLoader?

Reply:
Sorry, that is my mistake, I will make it right.